### ISSUE TITLE
Compensation : Comp Input Model

### ISSUE THREAD

**Issue by afs1jes** [afs1jes/CHNCT-Migration#10]

**Comment by csoroka** [comment]
Missing the Employee Name column, please add.  We want the employee name columns to still display on all the reports we just agreed to modify it to a text field so we had the ability to change the name with ease.  We discussed this over the phone in February when you were deciding to use the Actual LOB and Department (instead of the reserved members) to make all the reports and models a little easier to work with

**Comment by afs1jes** [comment]
I see Employee Names in column E.  If you don't see that, could you send me your built model?

**Comment by csoroka** [comment]
Christina Soroka has created a secure e-mail message for you at:

https://ssl.datamotion.com/r.aspx?b=16&eh=ckJ0z26Ut0iVoVYk411CGA&cb=chnct1

To access your message, simply follow these steps:

1. Click on the above link. 
2. You will be prompted to create a password to protect your account.
3. Access your Inbox to view your message.


If you are unable to open the link, please make sure that your firewall or your company's security settings are not blocking the link.


Additional Help:
•		Questions relating to the content of the e-mail should be directed to the sender or sending organization.
•		Questions relating to reading this secure message can be sent to helpdesk@chnct.org.


--------------------------------------------------------------------------------
Message Details: 
Subject: RE: [afs1jes/CHNCT-Migration] Compensation : Comp Input Model (#10)  
From: csoroka@chnct.org 
To: reply@reply.github.com 
Created: 4/27/2020  11:08:46 AM (UTC-04:00:00) 
Message Expires: 7/27/2020  11:08:46 AM (UTC-04:00:00)

Copyright (c) 1996-2020 DataMotion, Inc.

**Comment by csoroka** [comment]
I ran the Comp Input Model (2yr) for LOB 01 and Dept 200 (for 2020 WF) - I compared the data to the current system we are working off of and the data is the same.

**Comment by csoroka** [comment]
Compared one more time this morning and the 2 year model balanced to the current system model.  Wanted to see if the salary and tax dollars were properly linking to the input by account financial model but it doesn't look like those have been published yet.

**Comment by afs1jes** [comment]
Both of the Finance input models are published. They're in the "Financial Statements & Planning" application, in the "Revenue & Expense Input" grouping.  Both are visible to members of the role "Administrators" and I've confirmed that both you and Dina are in that role.... so you should see them.

Please confirm and let me know if you don't see them.

**Comment by csoroka** [comment]
I thought we were waiting on this one till Jim made the incentive changes (mentioned on our last call).  Unless that is already complete?  

Let me know.  Thank you,
Chrissy

**Comment by kclark-a3** [comment]
The incentive change was made by me in the .NET (new) and Jim made the change in current production system. Please continue with testing.

**Comment by csoroka** [comment]
Received an error when trying to submit for dept 01-103 - changed the start date on three positions and the 2022 incentive amount.
[Error comp input.pdf](https://github.com/afs1jes/CHNCT-Migration/files/9646202/Error.comp.input.pdf)

**Comment by AVHuntley** [comment]
Good catch. That error indicated that more than one value was being submitted to the same intersection. That was because the ReadOnly flags need to be in a different row in the new system.

I made that change - copied the ReadOnly flags from their original location to the new, correct one. I tested by opening the model and hitting 'Submit' without changing any values. Seemed to work, so hopefully problem is resolved.

**Comment by AVHuntley** [comment]
Formatted 'Latest Hire Date' and 'End Date' columns as MM/DD/YYYY

**Comment by csoroka** [comment]
Christina Soroka has created a secure email message for you at:
https://ssl.datamotion.com/r.aspx?b=16&eh=Iq8lwSai0kGABSUFDR-vdQ&cb=chnct1

To access your message, simply follow these steps:

1. Click on the above link. 
2. You will be prompted to create a password to protect your account.
3. Access your Inbox to view your message.

Problems accessing the message?
- If you are unable to open the link, please make sure that your firewall or your company's security settings are not blocking the link.
- If you are unsure of your password, please reset it at https://ssl.datamotion.com/ by clicking on the "Click HERE to reset your password" link.

Additional Help:
•		Questions relating to the content of the email should be directed to the sender or sending organization.
•		Questions relating to reading this secure message can be sent to ***@***.***


--------------------------------------------------------------------------------
Message Details: 
Subject: RE: [afs1jes/CHNCT-Migration] Compensation : Comp Input Model (#10) 
From: ***@***.*** 
To: ***@***.*** 
Message ID: 59682177
Created: 9/28/2022  7:02:02 AM (UTC-04:00:00) 
Expires: 12/28/2022  7:02:02 AM (UTC-04:00:00)
--------------------------------------------------------------------------------
Copyright (c) 1996-2022 DataMotion, Inc.

**Comment by csoroka** [comment]
Hello - 
It is possible to have this template display by Position # sort?

**Comment by csoroka** [comment]
Due to the 5yr comp input issue with the incentives - I updated the 2022 Annual incentive amounts in the 2yr model, to see if the same issue occurred (with the inceptives pro-rating) and it did.

-Chrissy

**Comment by afs1jes** [comment]
Input sheet row script changed to: 
ORDER(
Descendants([All Job Codes],,Leaves) *  
Descendants([All EmployeeStatus],,Leaves),
[JobCode].[JobCode].Currentmember.Name, basc)

This should give you the rows in Position name, ascending order.

**Comment by csoroka** [comment]
Thank you

**Comment by afs1jes** [comment]
Corrections made for 2 items:
1) The lookup from sheet "Summary", pulling data from sheet "Input" was still using rngInput instead of the new dataset named range, Dataset2. That has been changed in all columns of the Input sheet. This problem impacted all years in the model.

2) The Incentive annual amount was pulling from "Full Year", which worked fine until we got to this year where we have a partial zero year (actually, it was probably also a problem previously, but the impact was much smaller / less noticeable).  The input columns have been changed to interact with "Reserved Time" and I copied existing incentive amounts to that member.  The data for incentives now properly calculates regardless of previous submissions.  Note, however, that the amounts in 2022 Working Forecast may or may not be correct. They will not depreciate with every submission, but they may not be correct right now. When we refresh Comp Data again, this should be corrected.

**Comment by csoroka** [comment]
I went into the Comp input model for dept 01-200, to review the incentive amounts to ensure they didn't pro-rate when I reopened and all of the amounts that were keyed in prior are gone?  Do you know why they aren't appearing?

**Comment by AVHuntley** [comment]
I manually corrected some data by doing a 'refresh' from prod and copy/pasting within ad-hoc.

Because of the fix we made in October (where we adjusted the model to send incentive pay to [reserved time]), the refresh process didn't repopulate those cells correctly.

The data _should_ be corrected now. Let us know how it looks though.
![image](https://user-images.githubusercontent.com/88676502/205729662-c5efcaa0-e7f9-4e65-9788-348b70b0f305.png)

**Comment by csoroka** [comment]
I have updated all the incentive amounts for each department

**Comment by AVHuntley** [comment]
A lot of changes made to this and the 5yr templates. The goal was to have formulas resolve to "" instead of 0, when applicable. This should prevent sending values back to the cube that, when later read by templates, cause 'deleted' employees to still show up in future builds.

Current testing show parity between models in prod and dev, but it'd be good to keep an eye out for any strange behavior in this model in dev for a while.

**Comment by AVHuntley** [comment]
I altered the formula in the column for Annual Salary. Instead of:

=IF(**T28<>"0"**,T28,IF(R28<1,Q28,IF(OR(P28=0,P28="0"),Q28,P28)))

It's now:

=IF(**VALUE(T28)<>0**,T28,IF(R28<1,Q28,IF(OR(P28=0,P28="0"),Q28,P28)))

**Comment by csoroka** [comment]
Thank you Alec for updating the formula, I would like to do the comp copy again to test and move forward to testing both of the Comp input models.  I will do that this afternoon.  

-Chrissy

**Comment by csoroka** [comment]
I just completed my testing to ensure the salary information for 01-100, 01-101 & 01-200 tie to current production and it does.

Thank you
